src/main.rs

```diff
@@ -5,6 +5,10 @@
 #![feature(const_str_as_bytes)]
 #![feature(box_syntax)]
 #![feature(repeat_generic_slice)]
+#![feature(never_type)]
+
+#[macro_use]
+extern crate derive_more;
```

Comment:
This extern crate is probably needed for the `#[display(..)]` attrs?

Reply:
Yes.